### PR TITLE
Fixing #4980

### PR DESCRIPTION
**Pull request by mushthaq33**
Fixes https://github.com/wso2/product-apim/issues/4980

### CODE REVIEW

**Comment by pubudu538** [comment]
Fixing #4980 points to the 4980 issue in the carbon-apimgt. We should point to the product apim issue 4980.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java [hunk 1]
```diff
@@ -1883,4 +1897,41 @@ private boolean isAuthorizationFailure(Exception e) {
...
+     * @param mediation
+     * @return if the API is modified or not
+     */
+    private boolean isAPIModified(API api, Mediation mediation) {
+        if (mediation != null) {
+            String sequenceName;
+            if (APIConstants.API_CUSTOM_SEQUENCE_TYPE_IN.equalsIgnoreCase(mediation.getType())) {
+                sequenceName = api.getInSequence();
+                if (isSequenceExistsInAPI(sequenceName, mediation)) {
+                    api.setInSequence(null);
```
Comment (Rajith90) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java]: Why do we set null as the mediation and do an api update

Comment (mushthaq33) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java]: Is there any other way to remove the sequence which is already set? That's how we have done in other places as well I guess

Comment (rmsamitha) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java]: The same value of inSequence attribute of api object when an In sequence has not been set to an API when the API is created, should be set here too.

**Pull request merged**